src/main.rs

```diff
@@ -0,0 +1,282 @@
+use std::{
+    env,
+    io::{self, Write},
+    path::PathBuf,
+    str::FromStr,
+    time::Duration,
+};
+
+use reqwest::Url;
+use subalfred_core::state::fork_off::{fork_off, ForkOffConfig};
+use tokio::net::TcpListener;
+use tracing::{debug, trace};
+use zombienet_configuration::RegistrationStrategy;
+use zombienet_configuration::{types::AssetLocation, NetworkConfig};
+use zombienet_orchestrator::Orchestrator;
+use zombienet_orchestrator::{
+    metrics::{Metrics, MetricsHelper},
+    network::Network,
+    NetworkSpec, ScopedFilesystem,
+};
+use zombienet_provider::{
+    types::{RunCommandOptions, SpawnNodeOptions},
+    DynNamespace, DynNode, DynProvider, NativeProvider, Provider,
+};
+use zombienet_support::{fs::local::LocalFileSystem, net::wait_ws_ready};
+
+async fn sync(ns: DynNamespace) -> Result<(DynNode, String), ()> {
```

Comment:
This replaces the `poc.sh` script, correct?

Reply:
Yes, that correct 👍